src/sentry/event_manager.py

```diff
@@ -348,6 +369,8 @@ def save(self, project_id, raw=False, assume_normalized=False, start_time=None,
         _derive_plugin_tags_many(jobs, projects)
         _derive_interface_tags_many(jobs)
 
+        detect_synthetic_exception(job["event"])
```

Comment:
let's put this into _calculate_event_grouping, ideally the helper function lives in sentry.grouping too